### PR TITLE
Tests for mod-balances

### PR DESCRIPTION
**Pull request by NinoLipartiia**

### CODE REVIEW

File: contracts/mod-balances/Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,10 @@ cosmwasm-std = { workspace = true }
...
 cw2 = { workspace = true }
 cw20 = { workspace = true }
-thiserror = { workspace = true }
+cw20-base = { workspace = true }
+cw-multi-test = { workspace = true }
+cw-utils = { workspace = true }
 mod-sdk = { workspace = true }
+thiserror = { workspace = true }
 
 [dev-dependencies]
```
Comment (Buckram123) [inline contracts/mod-balances/Cargo.toml:28]:
```suggestion
mod-sdk = { workspace = true }
thiserror = { workspace = true }

[dev-dependencies]
cw20-base = { workspace = true }
cw-multi-test = { workspace = true }
cw-utils = { workspace = true }
```
We don't test dependencies to be included outside tests

File: contracts/mod-balances/src/msg.rs [hunk 2]
```diff
@@ -13,15 +13,12 @@ pub enum ExecuteMsg {}
...
-        denom: String,
-    },
+    GetBalance { address: String, denom: String },
     #[returns(QueryResponse)]
     GetCw20Balance {
         cw20_contract: String,
         address: String,
     },
     #[returns(QueryResponse)]
     HasBalanceComparator(HasBalanceComparator),
```
Comment (Buckram123) [inline contracts/mod-balances/src/msg.rs:23]: Let's add small descriptions to enum variants!

File: contracts/mod-balances/src/tests/balance_comparator.rs [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+            NATIVE_DENOM.to_string(),
+        ))),
+        comparator: BalanceComparator::Ne,
+    });
+    let res: QueryResponse = app
+        .wrap()
+        .query_wasm_smart(contract_addr.clone(), &msg)
+        .unwrap();
+    assert!(res.result);
+    assert_eq!(res.data, to_binary(&Uint128::new(1_000_000)).unwrap());
```
Comment (Buckram123) [inline contracts/mod-balances/src/tests/balance_comparator.rs:183]: Thanks for testing `QueryResponse.data`!

**Review by Buckram123 — APPROVED** [review]
Good job, few fixes and LGTM

File: contracts/mod-balances/src/tests/balance_comparator.rs [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+            NATIVE_DENOM.to_string(),
+        ))),
+        comparator: BalanceComparator::Lt,
+    });
+    let res: QueryResponse = app
+        .wrap()
+        .query_wasm_smart(contract_addr.clone(), &msg)
+        .unwrap();
+    assert!(!res.result);
+    assert_eq!(res.data, to_binary(&Uint128::new(1_000_000)).unwrap());
```
Comment (TrevorJTClarke) [inline contracts/mod-balances/src/tests/balance_comparator.rs:119]: Need a passing test for `Lt` after this

File: contracts/mod-balances/src/tests/balance_comparator.rs [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+    });
+    let res: QueryResponse = app
+        .wrap()
+        .query_wasm_smart(contract_addr.clone(), &msg)
+        .unwrap();
+    assert!(res.result);
+    assert_eq!(res.data, to_binary(&Uint128::from(15u128)).unwrap());
+
+    Ok(())
+}
```
Comment (TrevorJTClarke) [inline contracts/mod-balances/src/tests/balance_comparator.rs:353]: Same as with Native, needs coverage of `Gt` & `Gte`

**Review by TrevorJTClarke — CHANGES_REQUESTED** [review]
Hey @NinoLipartiia this is looking awesome!!
Left a couple comments on tests cov missing, minor things.

I think i made a big mistake on ergonomics as well.
The comparison is backwards, so for example the logic is currently:
`required_balance <= balance` 
which feels wrong, i believe it needs to be this for all of them:
`balance <= required_balance`
So when integrating, the user specifying Lte, expects the balance to fluctuate and the requirement to stay consistent.

Happy to chat on that if it doesn't make sense.

**Review by TrevorJTClarke — APPROVED** [review]
YEAHHHHHHHHHH

**Pull request merged**